enso/sample/__init__.py

```diff
@@ -0,0 +1,41 @@
+import numpy as np
+
+
+def sample(sampler, data, train_indices, train_size):
+    sampler = Sampler._class_for(sampler)(data, train_indices, train_size)
+    return sampler.sample()
+
+
+class Sampler(object):
+    """Base class for all samples."""
+
+    def __init__(self, data, train_indices, train_size):
+        self.data = data
+        self.train_indices = train_indices
+        self.train_size = train_size
+
+    def choose_random(self):
+        return np.random.choice(self.train_indices)
+
+    @property
+    def distances(self):
+        if not hasattr(self, '_distances'):
+            self._distances = pairwise_distances(self.points, metric=self.DISTANCE_FUNCTION)
+        return self._distances
+
+    @property
+    def points(self):
+        return [np.array(point) for point in self.data]
+
+    def sample(self):
+        raise NotImplementedError
+
+    @classmethod
+    def _class_for(cls, sampler_string):
+        from enso.sample.kcenter_sampler import KCenter
```

Comment:
Is there a better way to do this so that when you add a new sampler you don't need to add it as an import?

Reply:
I dunno if I would say "better" per se, but look at the `get_plugins` method. That's how we're handling it for other parts of the codebase. Basically it's dynamic imports where we just specify the name and take care of importing behind the scenes.